### PR TITLE
Drop older python version

### PR DESCRIPTION
**Pull request by mikemhenry**
## Description
We can drop python 3.8 testing per NEP 29
## Todos
- [ ] Implement feature / fix bug
- [ ] Add [tests](https://github.com/choderalab/openmmtools/tree/master/openmmtools/tests)
- [ ] Update [documentation](https://github.com/choderalab/openmmtools/tree/master/docs) as needed
- [ ] Update [changelog](https://github.com/choderalab/openmmtools/blob/master/docs/releasehistory.rst) to summarize changes in behavior, enhancements, and bugfixes implemented in this PR

## Status
- [ ] Ready to go

## Changelog message
```

```

### CODE REVIEW

**Comment by mikemhenry** [comment]
I added 3.11, if we have issues with solving we can remove it before merge.

**Comment by mikemhenry** [comment]
Looks like nose is unhappy

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10", "3.11"]
```
Comment (mikemhenry) [inline .github/workflows/CI.yml:27]:
```suggestion
        python-version: ["3.9", "3.10"]
```

**Comment by mikemhenry** [comment]
We can fix nose + py311 later

**Comment by mikemhenry** [comment]
If 3.11 works we will close this pr and merge in this one https://github.com/choderalab/openmmtools/pull/702 if 3.11 doesn't work, I will merge in this pr and close 702

**Review by ijpulidos — APPROVED** [review]
LGTM!

**Pull request merged**